Sources/MapboxDirections/Incident.swift

```diff
@@ -21,46 +20,174 @@ public struct Incident: Codable, Equatable {
+        self.shapeIndexRange = shapeIndexRange
+    }
+    
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        let formatter = ISO8601DateFormatter()
+        
+        identifier = try container.decode(String.self, forKey: .identifier)
+        rawKind = try container.decode(String.self, forKey: .type)
+        kind = Kind(rawValue: rawKind)
+        
+        description = try container.decode(String.self, forKey: .description)
+        
+        if let date = formatter.date(from: try container.decode(String.self, forKey: .creationDate)) {
+            creationDate = date
+        } else {
+            throw DecodingError.dataCorruptedError(forKey: .creationDate,
+                                                   in: container,
+                                                   debugDescription: "`Intersection.creationTime` is encoded with invalid format.")
+        }
+        if let date = formatter.date(from: try container.decode(String.self, forKey: .startDate)) {
+            startDate = date
+        } else {
+            throw DecodingError.dataCorruptedError(forKey: .startDate,
+                                                   in: container,
+                                                   debugDescription: "`Intersection.startTime` is encoded with invalid format.")
+        }
+        if let date = formatter.date(from: try container.decode(String.self, forKey: .endDate)) {
+            endDate = date
+        } else {
+            throw DecodingError.dataCorruptedError(forKey: .endDate,
+                                                   in: container,
+                                                   debugDescription: "`Intersection.endTime` is encoded with invalid format.")
+        }
+        
+        impact = try container.decodeIfPresent(String.self, forKey: .impact)
+        subtype = try container.decodeIfPresent(String.self, forKey: .subtype)
+        subtypeDescription = try container.decodeIfPresent(String.self, forKey: .subtypeDescription)
+        alertCodes = try container.decode(Set<Int>.self, forKey: .alertCodes)
+        
+        rawLanesBlocked = try container.decode(Set<String>.self, forKey: .lanesBlocked)
+        lanesBlocked = rawLanesBlocked.reduce(into: Set<BlockedLane?>()) { $0.insert(BlockedLane(rawValue: $1)) }
+        
+        let geometryIndexStart = try container.decode(Int.self, forKey: .geometryIndexStart)
+        let geometryIndexEnd = try container.decode(Int.self, forKey: .geometryIndexEnd)
+        shapeIndexRange = geometryIndexStart..<geometryIndexEnd
+    }
+    
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        let formatter = ISO8601DateFormatter()
+        
+        try container.encode(identifier, forKey: .identifier)
+        try container.encode(rawKind, forKey: .type)
```

Comment:
Do we need a separate `rawKind` property? I think we can make `Kind` conform to Codable and it’ll automatically be possible to replace `rawKind` with `kind` here.

Reply:
`rawKind` is introduced to preserve JSON encoding/decoding roundtrip for cases when current SDK version does not support new types of `Incident`s. Thus, we just store the raw value for coding purposes, and at the same time user is enabled to work only with known defined `Kind`s.

Same approach applied to `BlockedLane`, but IMHO it is less likely that new types of lanes could be introduced then new types of incidents. So I would rather agree that we should get rid of `rawLanesBlocked` instead.

Reply:
To avoid parallel properties, we should store only the raw value and make the enumeration-typed property a computed property.